Modules/QtWidgets/src/QmitkRenderWindowWidget.cpp

```diff
@@ -196,8 +198,8 @@ void QmitkRenderWindowWidget::InitializeGUI()
 
   connect(m_RenderWindow, &QVTKOpenGLWidget::mouseEvent, this, &QmitkRenderWindowWidget::MouseEvent);
 
-  mitk::TimeGeometry::ConstPointer timeGeometry = m_DataStorage->ComputeBoundingGeometry3D(m_DataStorage->GetAll());
-  mitk::RenderingManager::GetInstance()->InitializeViews(timeGeometry);
```

Comment:
Why did you remove this instead of using the code that you added in `RequestUpdate` and `ForceImmediateUpdate`:
```
auto geo = m_DataStorage->ComputeBoundingGeometry3D(m_DataStorage->GetAll());
mitk::RenderingManager::GetInstance()->InitializeView(m_RenderWindow->GetRenderWindow(), geo);
```

This will initialize the view for this specific render window.

Reply:
Here we have two problems:

1. `mitk::RenderingManager::GetInstance()->InitializeViews(timeGeometry);` has global scope, it creates the same problem that we have with:
`mitk::RenderingManager::GetInstance()->InitializeViewsByBoundingObjects(dataStorage);` 
See the example I described below. To avoid this I choose to use `InitializeView()` instead of `InitializeViews`. 

2. I moved the location to `RequestUpdate` and `ForceImmediateUpdate`, because in the case where data is loaded dynamically, lets say the user clicks an open file button, in this case `m_DataStorage` wouldn't have any data at `QmitkRenderWidget`'s construction time, in which case computing the geometry from an empty `m_DataStorage` wouldn't be necessary. This piece of code is actually useful if the path to the data set is specified through command line or has been hardcoded.

Reply:
I'm totally fine with 1.
I think 2. is a fix after changing the render window initialization of the `QmitkStdMultiWidget`: Here we do not re-initialize all render windows anymore but only the four render windows of the multi widget. So a single render window widget won't be affected. We need to be able to explicitly re-initialize a single render window widget, right?

Would it make sense to _request an update_ inside `InitializeGUI` in case we actually do have some data loaded in the data storage?

Reply:
> Would it make sense to request an update inside InitializeGUI in case we actually do have some data loaded in the data storage?

Yes it will make sense, I will add a geometry initialization inside the function.

> I think 2. is a fix after changing the render window initialization of the `QmitkStdMultiWidget`: 

Actually I explicitly initialized the geometries inside `QmitkStdMultiWidget`, would you prefer a call to `RequestUpdate` instead of `mitk::RenderingManager::GetInstance()->InitializeView(GetRenderWindow1()->GetRenderWindow(), geo);`? The result will be the same.

> Here we do not re-initialize all render windows anymore but only the four render windows of the multi widget. So a single render window widget won't be affected. We need to be able to explicitly re-initialize a single render window widget, right?

Yes this is the intention, although I ended up not calling `RequesUpdate`.